pandigital.rb

```diff
@@ -0,0 +1,39 @@
+# We shall say that an n-digit number is pandigital if it makes use of all the digits 1 to n exactly once;
+# for example, the 5-digit number, 15234, is 1 through 5 pandigital.
+
+# The product 7254 is unusual, as the identity, 39 × 186 = 7254, 
+# containing multiplicand, multiplier, and product is 1 through 9 pandigital.
+
+# Find the sum of all products whose multiplicand/multiplier/product identity can be written as a 1 through 9 pandigital.
+
+# HINT: Some products can be obtained in more than one way so be sure to only include it once in your sum.
+
+class Pandigital
+  def self.calculate
+    sum = 0
+    product_array = []
+    product = 0
+    (1..2000).each do |first_num|
+      (1..2000).each do |second_num|
+        # total length of multipliers must be 5 for a total length of 9 digits (figured out through trial and error)
+        if (first_num.to_s + second_num.to_s).length == 5
```

Comment:
What does the magic number `5` signify here? Perhaps making this a well-named constant would increase the clarity and intent?